addons/ai/functions/fnc_spawnWave.sqf

```diff
@@ -102,20 +120,37 @@ _logic setVariable ["Waves", (_wave-1)];
 _handlers = _logic getVariable ["Handlers",[]];
 {
     if(_x isEqualType {}) then {
-        [_wave,_spawnedGroups] call _x;
+        [_wave,_spawnedGroups,_spawnedUnits,_spawnedVehicles,_spawnedObjects,_logic] call _x;
     };
 } forEach _handlers;
 // Check if there is another wave
 if(_logic getVariable ["Waves",1] > 0) then {
+    private _time = _logic getvariable ["Time",10];
+    private _whenDead = _logic getVariable ["WhenDead",0];
 
-    // Check if we need to wait for them to die
-    if(_logic getVariable ["WhenDead",false]) then {
-        [{ {{alive _x} count (units _x) > 0 } count (_this select 1) <= 0 }, FUNC(spawnWave), [_logic,_spawnedGroups]] call CBA_fnc_waitUntilAndExecute;
-    }
-    else {  // Otherwise spawn the wave after sleeping for some time
-        [FUNC(spawnWave), [_logic], _logic getvariable ["Time",10]] call CBA_fnc_waitAndExecute;
-    };
+    // Wait for conditions before spawning waves
+    [
+        {
+            //params ["","","_minimumDead","_spawnedUnits", "_targetTime"];
```

Comment:
?

Reply:
Tiny bit of optimization, as otherwise it would have to execute `params []` on every frame. It is faster simply using select

Reply:
Sorry, i meant this is a pointless comment